add support for using an HTTPS proxy

rmi7 opened the pull request:
add support for using an HTTPS proxy for all requests,

- added new dependency `https-proxy-agent` to `package.json`
- added `proxy_url` default param to `lib/configure.js`
- added a check for a supplied `proxy_url` in `lib/client.js`, and if found uses the proxy through `https-proxy-agent`
- updated `README.md` with info on how to use this new `proxy_url` param

jaypatel512 commented:
Thanks @rmi7 for your PR. We will definitely look more into this PR, and merge once it is code reviewed.

rmi7 commented:
@jaypatel512 thanks for the response, based on your response I updated the PR

- I rebased on current `master`
- I removed some unwanted whitespace changes (because of default remove whitespace behaviour was enabled in my Atom)
- I tested and added more information in the updated README on how to use a username + password for the proxy.
- I generally improved my README changes to be more descriptive

I force pushed my updated branch to keep it clean!

looking forward to more feedback

rmi7 commented:
resolved the package.json conflict (uuid was removed, while I added https-proxy-agent)

rmi7 commented:
any chance this might be merged, otherwise I could just as well close the PR?

420Jessie78905 commented:
https://www.paypal.me/jessie4201     Jessiecuthbertson63@gmail.com if u can please send 6000 transfer to PayPal would be amazing no hydro or hot water it’s winter I got screwed for rent off the others please someone help me please the pay pal me is there as well as my email please and thank you
